backend/de.metas.payment.esr/src/main/java/de/metas/payment/esr/dataimporter/ESRImportEnqueuer.java

```diff
@@ -186,25 +271,81 @@ private void checkUpdateDataType(final I_ESR_Import esrImport, final String file
 			}
 			else
 			{
-				addLog("Assuming and updating type={} for ESR_Import={}", guessedType, esrImport);
-				esrImport.setDataType(guessedType);
-				save(esrImport);
+				addLog("Assuming and updating type={} for ESR_Import={}", guessedType, esrImportFile);
+				esrImportFile.setDataType(guessedType);
+				save(esrImportFile);
 			}
 		}
 		else
 		{
 			// see if the filename tells us that the user made a mistake
 			final String guessedType = ESRDataLoaderFactory.guessTypeFromFileName(fileName);
-			if (!Check.isEmpty(guessedType) && !guessedType.equalsIgnoreCase(esrImport.getDataType()))
+			if (!Check.isEmpty(guessedType) && !guessedType.equalsIgnoreCase(esrImportFile.getDataType()))
 			{
 				// throw error, telling the user to check the ESI_import's type
 				throw new AdempiereException(ESR_IMPORT_LOAD_FROM_FILE_INCONSITENT_TYPES);
 			}
 		}
 	}
 
-	private String computeESRHashAndCheckForDuplicates(final I_ESR_Import esrImport, final byte[] fileContent)
+	@NonNull
+	private File getUnzippedFile(final ZipInputStream zipStream,
```

Comment:
There is no point to write a file which you will have to read it again and discard it.
Better,
* rename the method as `extractResource`
* return org.springframework.core.io.Resource. Use org.springframework.core.io.ByteArrayResource implementation which basically will hold the "data" and the "filename" for you
* unzipDir and buffer params are not needed